Assets/Scripts/EOSPlayerDataStorageManager.cs

```diff
@@ -616,55 +400,32 @@ private void OnFileListRetrieved(ref QueryFileListCallbackInfo data)
             }
 
             SetFileList(fileNames);
-            foreach (var callback in FileListUpdateCallbacks)
-            {
-                callback?.Invoke();
-            }
+            OnFileListUpdated?.Invoke();
         }
 
         //-------------------------------------------------------------------------
         private ReadResult OnFileDataReceived(ref ReadFileDataCallbackInfo data)
         {
-            //if (data == null)
-            //{
-            //    Debug.LogError("Player Data Storage (OnFileDataReceived): data parameter is null!");
-            //    return ReadResult.FailRequest;
-            //}
-
-            return ReceiveData(data.Filename, data.DataChunk, data.TotalFileSizeBytes, data.IsLastChunk);
+            return ReceiveData(data.Filename, data.DataChunk, data.TotalFileSizeBytes) switch
```

Comment:
Whoa, I've never seen a switch statement like this. That's interesting. I learned some stuff by playing around with this in Visual Studio.

Reply:
Always fun to learn things :)